### PR TITLE
Bugfix/uxmcloud 1333

### PR DESCRIPTION
**Pull request by whispyy**
### Description of the Change

- error and warning message are now stackable
- warning message can be dismiss

### CODE REVIEW

File: client/app/common/cloud-message.service.js [hunk 1]
```diff
@@ -37,47 +36,47 @@ class CloudMessage {
         }
     }
 
-    getMessageHandler (stateName) {
+    getMessageHandler (containerName) {
```
Comment (neolitec) [inline client/app/common/cloud-message.service.js]: I think we need some jsdoc for public methods of transversal services :)

File: client/components/me-alerts/me-alerts.controller.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class MeAlertsCtrl {
```
Comment (neolitec) [inline client/components/me-alerts/me-alerts.controller.js]: What's the meaning of `MeAlerts`?

Comment (Mboulianne) [inline client/components/me-alerts/me-alerts.controller.js]: I thought the same think... MeAlerts don't say anything about the purpose of the controller / page / service, etc.

Comment (Mboulianne) [inline client/components/me-alerts/me-alerts.controller.js]: but it was already in place (page + service) and we tend to name controller based on the associated page name.

File: client/components/me-alerts/me-alerts.service.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        this.$translatePartialLoader = $translatePartialLoader.addPart("../components/me-alerts");
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+        this.UserAlertsAapi = UserAlertsAapi;
+        this.REDIRECT_URLS = REDIRECT_URLS;
+        this.TARGET = TARGET;
+    }
+
+    getMessages () {
+        this.$translate.refresh().then(() => this.UserAlertsAapi.query({
```
Comment (Mboulianne) [inline client/components/me-alerts/me-alerts.service.js]: I'm curious about this $translate.refresh() thingy.. what does it accomplish?

File: client/components/me-alerts/me-alerts.controller.js (hunk 2 above)
Comment (whispyy) [inline client/components/me-alerts/me-alerts.controller.js]: Yap, I had in mind to keep the current structure in the way to respect consistency between managers.
Following telecom manager : https://github.com/ovh-ux/ovh-manager-telecom/tree/master/client/components/me-alerts

File: client/app/common/cloud-message.service.js (hunk 1 above)
Comment (whispyy) [inline client/app/common/cloud-message.service.js]: Oh yes I totally agree with you

File: client/components/me-alerts/me-alerts.service.js (hunk 3 above)
Comment (Netnix) [reply client/components/me-alerts/me-alerts.service.js]: It's to make sure the translation tool is ready. That way we don't have "cloud_machin_chouette" labels across the manager. We used to have this issue a while ago.

File: client/app/ui-components/message/message-container.component.js [hunk 4]
```diff
@@ -1,66 +1,111 @@
...
+    }
+
     class CuiMessageContainerCtrl {
         constructor ($scope) {
             this.$scope = $scope;
             this.messages = this.messages || [];
         }
 
         $onInit () {
+            if (!this.dismissableTypes) {
```
[the remark this answers is not in the document]
Comment (whispyy) [reply client/app/ui-components/message/message-container.component.js]: seems to be impossible to pass in parameters this.smth.

File: client/components/me-alerts/me-alerts.service.js (hunk 3 above)
Comment (Mboulianne) [reply client/components/me-alerts/me-alerts.service.js]: That's weird... the translator is initiated in a config and services are instantiated AFTER all config ran.

File: client/app/ui-components/message/message-container.component.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Mboulianne) [reply client/app/ui-components/message/message-container.component.js]: now that I'm looking at it => 

this.dismissableTypes = this.dismissableTypes || ["info", "success"];

less nesting = me happy

use the same strategy for groupedTypes.

[the remark this answers is not in the document]
Comment (Mboulianne) [reply client/app/ui-components/message/message-container.component.js]: I would try this if I were you 

this.dismissableTypes = this.dismissableTypes || ["info", "success"];

less nesting = me happy

use the same strategy for groupedTypes.

[the remark this answers is not in the document]
Comment (whispyy) [reply client/app/ui-components/message/message-container.component.js]: done

**Pull request merged**